pymodule.py

```diff
@@ -38,101 +38,91 @@
+                   'This is not the best for multireference computations.',
+                   'To use Psi4 CASSCF orbitals, set REF_TYPE to CASSCF.']
+            msg = '\n  '.join(msg)
+            warnings.warn(f"\n  {msg}\n", UserWarning)
+
+        wfn = psi4.driver.scf_helper('forte', molecule=molecule, **kwargs)
+    elif ref_type in ['casscf', 'rasscf']:
+        wfn = psi4.proc.run_detcas(ref_type, molecule=molecule, **kwargs)
     else:
-        return mo_space_info
-
+        raise ValueError(f"Invalid REF_TYPE: {ref_type.upper()} not available!")
 
-def prepare_forte_objects_from_psi4_wfn(options, wfn, mo_space_info):
-    """
-    Take a psi4 wavefunction object and prepare the ForteIntegrals, SCFInfo, and MOSpaceInfo objects
+    return wfn
 
-    Parameters
-    ----------
-    options : ForteOptions
-        A Forte ForteOptions object
-    wfn : psi4 Wavefunction
-        A psi4 Wavefunction object
-    mo_space_info : the MO space info read from options
-        A Forte MOSpaceInfo object
 
-    Returns
-    -------
-    tuple(ForteIntegrals, SCFInfo, MOSpaceInfo)
-        a tuple containing the ForteIntegrals, SCFInfo, and MOSpaceInfo objects
+def check_MO_overlap(options, molecule, Ca):
     """
+    Test the MO overlp matrix is identity or not.
+    We will recompute the SO overlap from molecule and test against Ca,
+    S_MO = Ca^T S_SO Ca.
+    If the molecule and Ca are consistent, S_MO should be identity matrix.
 
-    # Call methods that project the orbitals (AVAS, embedding)
-    mo_space_info = orbital_projection(wfn, options, mo_space_info)
+    :param options: a ForteOptions object
+    :param molecule: a Psi4 Molecule object
+    :param Ca: a Psi4 Matrix that holds orbital coefficients
 
-    # Build Forte SCFInfo object
-    scf_info = forte.SCFInfo(wfn)
-
-    # Build a map from Forte StateInfo to the weights
-    state_weights_map = forte.make_state_weights_map(options, mo_space_info)
-
-    return (state_weights_map, mo_space_info, scf_info)
+    :return: a tuple of (S_MO == I, a new Psi4 Wavefunction object)
+    """
+    p4print = psi4.core.print_out
+    p4print("\n\n")
+
+    basis = options.get_str('BASIS')
+    wfn = psi4.core.Wavefunction.build(molecule, basis)
+
+    if psi4.core.get_global_option("RELATIVISTIC") in ["X2C", "DKH"]:
+        basis_rel = options.get_str("BASIS_RELATIVISTIC")
+        puream = wfn.basisset().has_puream()
+        rel_bas = psi4.core.BasisSet.build(molecule, "BASIS_RELATIVISTIC",
+                                           basis_rel, "DECON", basis,
+                                           puream=puream)
+        wfn.set_basisset('BASIS_RELATIVISTIC', rel_bas)
+
+    p4print("\n  Checking orbital orthonormality against current geometry ...")
+
+    # build MO overlap
+    mints = wfn.mintshelper()
+    S = mints.so_overlap()
+    mo_overlap = psi4.core.triplet(Ca, S, Ca, True, False, False)
+
+    # test orbital orthonormality
+    mo_overlap.zero_diagonal()
```

Comment:
This checks orthogonality, not orthonormality. Checking orthogonality should be enough for the orbital reading errors we're worried about, but it's still different from orthonormality.

Reply:
You are correct, let me change the wording.

Reply:
OK, I decided to change the function instead.